easybuild/easyblocks/l/llvm.py

```diff
@@ -497,8 +497,22 @@ def _configure_final_build(self):
             self._cmakeopts['LIBOMP_HWLOC_INSTALL_DIR'] = hwloc_root
 
         if 'openmp' in self.final_projects:
-            if LooseVersion(self.version) >= LooseVersion('19') and self.cfg['build_openmp_offload']:
-                self.runtimes_cmake_args['LIBOMPTARGET_PLUGINS_TO_BUILD'] = '%s' % '|'.join(self.offload_targets)
+            if self.cfg['build_openmp_offload']:
+                # Force dlopen of the GPU libraries at runtime, not using existing libraries
+                if LooseVersion(self.version) >= LooseVersion('19'):
+                    self.runtimes_cmake_args['LIBOMPTARGET_PLUGINS_TO_BUILD'] = '%s' % '|'.join(self.offload_targets)
+                    dlopen_plugin_list = []
+                    if self.amdgpu_target_cond:
+                        dlopen_plugin_list += ['amdgpu']
+                    if self.nvptx_target_cond:
+                        dlopen_plugin_list += ['cuda']
+                    if dlopen_plugin_list:
+                        self._cmakeopts['LIBOMPTARGET_DLOPEN_PLUGINS'] = "'%s'" % ';'.join(dlopen_plugin_list)
```

Comment:
I think this is good to go.
Only thing I am wondering is if it would be worth to do something like
```python

OMPT_DLOPEN_PLUGINS = ['cuda', 'amdgpu']
...
...
                    dlopen_plugin_list = set(self.offload_targets) & set(OMPT_DLOPEN_PLUGINS)
```